bindings-libcddb no longer compiles with GCC 14

GCC 14 [changed some warnings to errors](https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors). As a result `bindings-libcddb` no longer builds (this may also affect other packages here, maybe good idea to double check).

Full log:

```
Running phase: setupCompilerEnvironmentPhase
Build with /nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6.
Running phase: unpackPhase
unpacking source archive /nix/store/p18dj45dvrmmwip2rz4x3dj1w36f8s6q-bindings-libcddb-0.3.tar.gz
source root is bindings-libcddb-0.3
setting SOURCE_DATE_EPOCH to timestamp 1277032434 of file bindings-libcddb-0.3/src/inlines.c
Running phase: patchPhase
Running phase: compileBuildDriverPhase
setupCompileFlags: -package-db=/build/tmp.BnlqXWN1Fl/setup-package.conf.d -threaded
[1 of 2] Compiling Main             ( Setup.hs, /build/tmp.BnlqXWN1Fl/Main.o )
[2 of 2] Linking Setup
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
configureFlags: --verbose --prefix=/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3 --libdir=$prefix/lib/$compiler/lib --libsubdir=$abi/$libname --docdir=/nix/store/ywxnkp0258m6k79dk730v5rn95ldb82r-bindings-libcddb-0.3-doc/share/doc/bindings-libcddb-0.3 --with-gcc=gcc --package-db=/build/tmp.BnlqXWN1Fl/package.conf.d --ghc-option=-j2 --ghc-option=+RTS --ghc-option=-A64M --ghc-option=-RTS --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --enable-split-sections --enable-library-stripping --enable-executable-stripping --ghc-option=-haddock --extra-lib-dirs=/nix/store/wm1qn5jqrxpcjkc640gq8a90ns5gw3cn-ncurses-6.4.20221231/lib --extra-lib-dirs=/nix/store/r04wysx6yl7ys1rdv1b1lih6in1810r0-libffi-3.4.6/lib --extra-lib-dirs=/nix/store/wkysiv0sxqs6wn44nb46bbyig4plzsy4-elfutils-0.192/lib --extra-lib-dirs=/nix/store/ghb61mqc90qf1nx6kyahhhagicyxvcyg-gmp-with-cxx-6.3.0/lib --extra-include-dirs=/nix/store/xkcig2php7ddrx2lna7vdlaqndgcl1zs-libcddb-1.3.2/include --extra-lib-dirs=/nix/store/xkcig2php7ddrx2lna7vdlaqndgcl1zs-libcddb-1.3.2/lib --extra-include-dirs=/nix/store/ym52wgq3kyp6c3sy27qnzd4g81i4d5v6-glibc-iconv-2.40/include
Using Parsec parser
Warning: bindings-libcddb.cabal:19:16: version operators used. To use version
operators the package needs to specify at least 'cabal-version: >= 1.8'.
Warning: bindings-libcddb.cabal:20:27: version operators used. To use version
operators the package needs to specify at least 'cabal-version: >= 1.8'.
Configuring bindings-libcddb-0.3...
Dependency base >=3 && <5: using base-4.18.2.1
Dependency bindings-DSL >=1.0 && <1.1: using bindings-DSL-1.0.25
Dependency libcddb >=1.3.2: using version 1.3.2
Source component graph: component lib
Configured component graph:
    component bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG
        include base-4.18.2.1
        include bindings-DSL-1.0.25-D82K1L2M3QGLj7IZe182jE
Linked component graph:
    unit bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG
        include base-4.18.2.1
        include bindings-DSL-1.0.25-D82K1L2M3QGLj7IZe182jE
        Bindings.Libcddb=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb,Bindings.Libcddb.Cddb=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.Cddb,Bindings.Libcddb.CddbCmd=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbCmd,Bindings.Libcddb.CddbConn=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbConn,Bindings.Libcddb.CddbDisc=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbDisc,Bindings.Libcddb.CddbError=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbError,Bindings.Libcddb.CddbLog=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbLog,Bindings.Libcddb.CddbSite=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbSite,Bindings.Libcddb.CddbTrack=bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG:Bindings.Libcddb.CddbTrack
Ready component graph:
    definite bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG
        depends base-4.18.2.1
        depends bindings-DSL-1.0.25-D82K1L2M3QGLj7IZe182jE
Using Cabal-3.10.3.0 compiled by ghc-9.6
Using compiler: ghc-9.6.6
Using install prefix:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3
Executables installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/bin
Libraries installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/lib/ghc-9.6.6/lib/x86_64-linux-ghc-9.6.6/bindings-libcddb-0.3-J3ZQPXC8bh37rsZQgwj6KG
Dynamic Libraries installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/lib/ghc-9.6.6/lib/x86_64-linux-ghc-9.6.6
Private executables installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/libexec/x86_64-linux-ghc-9.6.6/bindings-libcddb-0.3
Data files installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/share/x86_64-linux-ghc-9.6.6/bindings-libcddb-0.3
Documentation installed in:
/nix/store/ywxnkp0258m6k79dk730v5rn95ldb82r-bindings-libcddb-0.3-doc/share/doc/bindings-libcddb-0.3
Configuration files installed in:
/nix/store/0yxi1w8srh085cj2rmsx6wg59hby1idi-bindings-libcddb-0.3/etc
No alex found
Using ar found on system at:
/nix/store/srcmmqi8kxjfygd0hyy42c8hv6cws83b-binutils-wrapper-2.43.1/bin/ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 14.2.1 given by user at:
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc
Using ghc version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/ghc
Using ghc-pkg version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/ghc-pkg-9.6.6
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.29.2 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/haddock-ghc-9.6.6
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/hpc-ghc-9.6.6
Using hsc2hs version 0.68.9 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/hsc2hs-ghc-9.6.6
Using hscolour version 1.25 found on system at:
/nix/store/v46cywpacz0q1snif3ac0crw31pq3h6z-hscolour-1.25/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/srcmmqi8kxjfygd0hyy42c8hv6cws83b-binutils-wrapper-2.43.1/bin/ld.gold
Using pkg-config version 0.29.2 found on system at:
/nix/store/0d4m43yp69lrm8imxbqgl9zxjwwz52jw-pkg-config-wrapper-0.29.2/bin/pkg-config
Using runghc version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/runghc-9.6.6
Using strip version 2.43 found on system at:
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/strip
Using tar found on system at:
/nix/store/9cwwj1c9csmc85l2cqzs3h9hbf1vwl6c-gnutar-1.35/bin/tar
No uhc found
Running phase: buildPhase
Preprocessing library for bindings-libcddb-0.3..
Building library for bindings-libcddb-0.3..
[1 of 9] Compiling Bindings.Libcddb.Cddb ( dist/build/Bindings/Libcddb/Cddb.hs, dist/build/Bindings/Libcddb/Cddb.o, dist/build/Bindings/Libcddb/Cddb.dyn_o )
[2 of 9] Compiling Bindings.Libcddb.CddbError ( dist/build/Bindings/Libcddb/CddbError.hs, dist/build/Bindings/Libcddb/CddbError.o, dist/build/Bindings/Libcddb/CddbError.dyn_o )
[3 of 9] Compiling Bindings.Libcddb.CddbLog ( dist/build/Bindings/Libcddb/CddbLog.hs, dist/build/Bindings/Libcddb/CddbLog.o, dist/build/Bindings/Libcddb/CddbLog.dyn_o )
[4 of 9] Compiling Bindings.Libcddb.CddbSite ( dist/build/Bindings/Libcddb/CddbSite.hs, dist/build/Bindings/Libcddb/CddbSite.o, dist/build/Bindings/Libcddb/CddbSite.dyn_o )
[5 of 9] Compiling Bindings.Libcddb.CddbTrack ( dist/build/Bindings/Libcddb/CddbTrack.hs, dist/build/Bindings/Libcddb/CddbTrack.o, dist/build/Bindings/Libcddb/CddbTrack.dyn_o )
[6 of 9] Compiling Bindings.Libcddb.CddbDisc ( dist/build/Bindings/Libcddb/CddbDisc.hs, dist/build/Bindings/Libcddb/CddbDisc.o, dist/build/Bindings/Libcddb/CddbDisc.dyn_o )
[7 of 9] Compiling Bindings.Libcddb.CddbConn ( dist/build/Bindings/Libcddb/CddbConn.hs, dist/build/Bindings/Libcddb/CddbConn.o, dist/build/Bindings/Libcddb/CddbConn.dyn_o )
[8 of 9] Compiling Bindings.Libcddb.CddbCmd ( dist/build/Bindings/Libcddb/CddbCmd.hs, dist/build/Bindings/Libcddb/CddbCmd.o, dist/build/Bindings/Libcddb/CddbCmd.dyn_o )
[9 of 9] Compiling Bindings.Libcddb ( src/Bindings/Libcddb.hs, dist/build/Bindings/Libcddb.o, dist/build/Bindings/Libcddb.dyn_o )
[1 of 9] Compiling Bindings.Libcddb.Cddb ( dist/build/Bindings/Libcddb/Cddb.hs, dist/build/Bindings/Libcddb/Cddb.p_o )
[2 of 9] Compiling Bindings.Libcddb.CddbError ( dist/build/Bindings/Libcddb/CddbError.hs, dist/build/Bindings/Libcddb/CddbError.p_o )
[3 of 9] Compiling Bindings.Libcddb.CddbLog ( dist/build/Bindings/Libcddb/CddbLog.hs, dist/build/Bindings/Libcddb/CddbLog.p_o )
[4 of 9] Compiling Bindings.Libcddb.CddbSite ( dist/build/Bindings/Libcddb/CddbSite.hs, dist/build/Bindings/Libcddb/CddbSite.p_o )
[5 of 9] Compiling Bindings.Libcddb.CddbTrack ( dist/build/Bindings/Libcddb/CddbTrack.hs, dist/build/Bindings/Libcddb/CddbTrack.p_o )
[6 of 9] Compiling Bindings.Libcddb.CddbDisc ( dist/build/Bindings/Libcddb/CddbDisc.hs, dist/build/Bindings/Libcddb/CddbDisc.p_o )
[7 of 9] Compiling Bindings.Libcddb.CddbConn ( dist/build/Bindings/Libcddb/CddbConn.hs, dist/build/Bindings/Libcddb/CddbConn.p_o )
[8 of 9] Compiling Bindings.Libcddb.CddbCmd ( dist/build/Bindings/Libcddb/CddbCmd.hs, dist/build/Bindings/Libcddb/CddbCmd.p_o )
[9 of 9] Compiling Bindings.Libcddb ( src/Bindings/Libcddb.hs, dist/build/Bindings/Libcddb.p_o )

In file included from src/inlines.c:1:0: error: 
src/inlines.c: In function ‘array_CDDB_CATEGORY’:

src/inlines.c:5:16: error:
     error: returning ‘const char **’ from a function with incompatible return type ‘char * const*’ [-Wincompatible-pointer-types]
        5 | BC_GLOBALARRAY(CDDB_CATEGORY,char*)
          |                ^~~~~~~~~~~~~
  |
5 | BC_GLOBALARRAY(CDDB_CATEGORY,char*)
  |                ^

/nix/store/h7gyv6j48hp3jk0nm5gb29czkb3yjbmr-bindings-DSL-1.0.25/lib/ghc-9.6.6/lib/x86_64-linux-ghc-9.6.6/bindings-DSL-1.0.25-D82K1L2M3QGLj7IZe182jE/include/bindings.cmacros.h:14:10: error:
     note: in definition of macro ‘BC_GLOBALARRAY’
       14 |   return name; \
          |          ^~~~
   |
14 |   return name; \
   |          ^
`gcc' failed in phase `C Compiler'. (Exit code: 1)
```

--- a/bindings-libcddb/src/inlines.c
+++ b/bindings-libcddb/src/inlines.c
@@ -2,4 +2,4 @@
 #include <cddb/cddb.h>
 
 BC_INLINE1(SEARCHCAT,cddb_cat_t,cddb_cat_t)
-BC_GLOBALARRAY(CDDB_CATEGORY,char*)
+BC_GLOBALARRAY(CDDB_CATEGORY,const char*)

